leapp/utils/report.py

```diff
@@ -1,72 +1,20 @@
 import json
-import six
-
-from jinja2 import Template
 
 from leapp.utils.audit import get_messages
 
 
-def _flatten_dict(dict_):
-    new_dict = {}
-    if type(dict_) is not dict:
-        return new_dict
-
-    for k, v in six.iteritems(dict_):
-        if type(v) == dict:
-            new_dict.update(_flatten_dict(v))
-        else:
-            new_dict[k] = v
-
-    return new_dict
-
-
-def fetch_upgrade_report_raw(context_id, renderers=True):
+def fetch_upgrade_report_messages(context_id):
     """
     :param context_id: ID to identify the needed messages
     :type context_id: str
-    :param renderers: whether copy or skip renderers field of original message
-    :type boolean
-    :return: A list of upgrade messages of type "Report" withing the given context,
-             each message is a dict of raw data.
+    :return: All upgrade messages of type "Report" withing the given context
     """
     report_msgs = get_messages(names=['Report'], context=context_id) or []
 
     messages = []
     for message in report_msgs:
-        data = json.loads(message.get('message').get('data'))
-        merged_data = {
-            'topic': message.get('topic'),
-            'phase': message.get('phase'),
-            'audience': data['audience'],
-            'title': data['title'],
-            'flags': data['flags'],
-            'severity': data['severity']
-        }
-        if renderers:
-            merged_data['renderers'] = data['renderers']
-        detail = _flatten_dict(json.loads(data['detail']))
-        merged_data.update(detail)
-        messages.append(merged_data)
+        data = message.get('message').get('data')
```

Comment:
Open question: do we sanitize message data somewhere? In case like message = {} this will raise AttributeError.

Reply:
I see https://github.com/oamg/leapp/blob/04e6041faf0e7852c37eff069fe6468ce23d8fae/leapp/utils/audit/__init__.py#L393

but not sure how much this covers us. @vinzenz what do you think?

Reply:
This particular case is covered now - one shouldn't be able to get a message without data member. If so, the problem lays deeper. Actually for the intend one should probably make this:
```python
data = message['message']['data']
```
which would rather give you a KeyError instead.

Reply:
changed it to `data = message['message']['data']` apart from that, as discussed, we consider this ok...